# 통계분석 과제 
## df_Xtrain= pd.read_csv('X_train.csv',encoding = 'cp949')
## 1 . X_train 데이터에서 강남점과 일산점의 총구매액 평균의 차이를 검정하세요 (정규성은 만족한다고 가정) 


In [1]:
import pandas as pd 
df_Xtrain= pd.read_csv('./data/X_train.csv',encoding = 'cp949')
df_Xtrain

cust_id       총구매액     최대구매액       환불금액   주구매상품 주구매지점  내점일수   내점당구매건수  \
0           0   68282840  11264000  6860000.0      기타   강남점    19  3.894737   
1           1    2136000   2136000   300000.0     스포츠   잠실점     2  1.500000   
2           2    3197000   1639000        NaN  남성 캐주얼   관악점     2  2.000000   
3           3   16077620   4935000        NaN      기타   광주점    18  2.444444   
4           4   29050000  24000000        NaN      보석  본  점     2  1.500000   
...       ...        ...       ...        ...     ...   ...   ...       ...   
3495     3495    3175200   3042900        NaN      골프  본  점     1  2.000000   
3496     3496   29628600   7200000  6049600.0    시티웨어  부산본점     8  1.625000   
3497     3497      75000     75000        NaN    주방용품   창원점     1  1.000000   
3498     3498    1875000   1000000        NaN     화장품  본  점     2  1.000000   
3499     3499  263101550  34632000  5973000.0      기타  본  점    38  2.421053   

        주말방문비율  구매주기  
0     0.527027    17  
1     0.000000     1  
2     0.000000     1  
3     0.318182    16  
4     0.000000    85  
...        ...   ...  
3495  1.000000     0  
3496  0.461538    40  
3497  0.000000     0  
3498  0.000000    39  
3499  0.467391     8  

[3500 rows x 10 columns]

In [4]:
# 주구매지점 확인
df_Xtrain['주구매지점'].value_counts()

본  점     1077
잠실점       474
분당점       436
부산본점      245
영등포점      241
일산점       198
강남점       145
광주점       114
노원점        90
청량리점       86
대전점        70
미아점        69
부평점        57
동래점        49
관악점        46
인천점        34
안양점        29
포항점        11
대구점         7
센텀시티점       6
울산점         6
전주점         5
창원점         4
상인점         1
Name: 주구매지점, dtype: int64

In [5]:
# 강남점, 일산점 추출하기
df_gangnam = df_Xtrain.loc[df_Xtrain['주구매지점']=='강남점', '총구매액']
df_ilsan = df_Xtrain.loc[df_Xtrain['주구매지점']=='일산점', '총구매액']

In [7]:
import scipy.stats as stats

# 등분산성 검정하기
# 귀무가설 : 등분산성을 만족한다.
# 대립가설 : 등분산성을 만족하지 않는다.
stats.levene(df_gangnam, df_ilsan)

LeveneResult(statistic=4.984728737913695, pvalue=0.02622071970102609)

### 결과 해석
__1.__ p-value값이 유의수준(0.05)보다 작으므로 귀무가설을 기각한다,
- 등분산성을 만족하지 않는다.

In [8]:
# t-test실시하기
stats.ttest_ind(df_gangnam, df_ilsan, equal_var=False)

Ttest_indResult(statistic=-2.0905294693218486, pvalue=0.03746469304567647)

### 결과 해석
__H0__ : 강남점과 일산점의 총구매액 평균의 차이는 없다(유의하지 않다)
__Ha__ : 강잠점과 일산점의 총구매액 평균의 차이는 유의하다.
__1.__ p-value값(0.037)이 유의수준(0.05)보다 작으므로 귀무가설을 기각한다.
- 강남점과 일산점의 총구매액 평균 차이는 유의하다.

### 시각화